src/drivers/cmakeServerDriver.ts

```diff
@@ -63,15 +64,55 @@ export class CMakeServerDriver extends CMakeDriver {
     private _prevConfigureEnv = 'null';
 
     // TODO: Refactor to make this assertion unecessary
-    private _codeModel!: null | cms.CodeModelContent;
-    get codeModel(): null | cms.CodeModelContent {
-        return this._codeModel;
-    }
-    set codeModel(v: null | cms.CodeModelContent) {
-        this._codeModel = v;
+    private codeModel!: CodeModelContent | null;
+    private convertServerCodeModel(serverCodeModel: null | ServerCodeModelContent): CodeModelContent | null {
+        if (serverCodeModel) {
+            const codeModel: CodeModelContent = { configurations: [], toolchains: undefined};
+            for (const config of serverCodeModel.configurations) {
+                const newConfig: CodeModelConfiguration = { name: config.name, projects: [] };
+                for (const project of config.projects) {
+                    const newProject: CodeModelProject = {
+                        name: project.name,
+                        sourceDirectory: project.sourceDirectory,
+                        hasInstallRule: project.hasInstallRule,
+                        targets: []
+                    };
+                    for (const target of project.targets) {
+                        const newTarget: CodeModelTarget = {
+                            name: target.name,
+                            type: target.type,
+                            sourceDirectory: target.sourceDirectory,
+                            fullName: target.fullName,
+                            artifacts: target.artifacts,
+                            sysroot: target.sysroot,
+                            fileGroups: []
+                        };
+                        if (target.fileGroups) {
+                            newTarget.fileGroups = [];
+                            for (const group of target.fileGroups) {
+                                const newGroup: CodeModelFileGroup = {
+                                    sources: group.sources,
+                                    language: group.language,
+                                    includePath: group.includePath,
+                                    defines: group.defines,
+                                    isGenerated: group.isGenerated,
+                                    compileCommandFragments: group.compileFlags ? [...shlex.split(group.compileFlags)] : []
```

Comment:
we should populate the correct one (flags, or fragments) based on the API version.   So this code will need to know the API version somehow.  Maybe initialize the driver with it?  Or get it from extension.ts?